tools/mk_vendor

```diff
@@ -0,0 +1,854 @@
+    ${NAME} --version
+
+build and interact with vendor libraries
+
+Commands:
+    Get help for commands with ${NAME} COMMAND --help
+
+    make    build the vendor library
+    locate  find a vendor library
+
+Common Options:
+    -h, --help  display help and exit
+    --version   display version and exit
+
+Examples:
+    build the default version of OpenSSL:
+      \$ ${NAME} make openssl
+
+    build OpenSSL 1.1.1u:
+      \$ ${NAME} make openssl:openssl111u
+
+    locate the prefix directory of OpenSSL 3.1.2 with asan
+      \$ ${NAME} locate openssl:openssl312 --options=asan
+"
+
+MAKE_USAGE="Usage:
+    ${NAME} make <configuration>[:<preset>] [vendor-options]
+    ${NAME} make -h | --help
+
+fetch, patch, build and install a vendor library
+
+Vendor Options:
+    -n,--name=STRING       override the configuration name
+    -f,--fetch=STRING      override the fetch method used to retrieve the sources
+    -p,--patch=STRING,...  override the patch sets applied to the sources
+    -b,--build=STRING,...  override the build scripts configuration
+    --options=STRING,...   override the build options
+
+Other Options:
+    -h, --help  display help and exit
+    --dry-run   stop after reading the configuration file
+
+Examples:
+    build the default version of OpenSSL:
+      \$ ${NAME} make openssl
+
+    build OpenSSL 1.1.1u with patched CVE and ASAN:
+      \$ ${NAME} make openssl:openssl111u --patch=all_CVE --options=asan
+
+    build OpenSSL 1.1.1j and give it a different name:
+      \$ ${NAME} make openssl:openssl111j --name=openssl-default
+"
+
+LOCATE_USAGE="Usage:
+    ${NAME} locate <configuration>[:<preset>] [vendor-options]
+    ${NAME} locate -h | --help
+
+Display the path to the install prefix that would be created by passing the same
+vendor configuration to '${NAME} make'.
+
+Vendor Options:
+    -n,--name=STRING       override the configuration name
+    -f,--fetch=STRING      override the fetch method used to retrieve the sources
+    -p,--patch=STRING,...  override the patch sets applied to the sources
+    -b,--build=STRING,...  override the build scripts configuration
+    --options=STRING,...   override the build options
+
+Other Options:
+    -h, --help  display help and exit
+"
+
+
+die() {
+    printf 'error: %s: %s\n' "${NAME}" "$1" >&2
+    exit 1
+}
+
+fatal() {
+    printf 'error: %s internal error: %s\n' "${NAME}" "$1" >&2
+    exit 2
+}
+
+need_cmd() {
+    if ! command -v "$1" > /dev/null 2>&1; then
+        die "required command not found: '$1'"
+    fi
+}
+
+normalize_path() {
+    if command -v readlink > /dev/null 2>&1; then
+        if readlink -f "$1" > /dev/null 2>&1; then
+            readlink -f "$1" 2> /dev/null
+            return
+        fi
+    fi
+
+    if command -v perl > /dev/null 2>&1; then
+        if perl -MCwd -le 'print Cwd::abs_path shift' "$1" > /dev/null 2>&1; then
+            perl -MCwd -le 'print Cwd::abs_path shift' "$1" 2> /dev/null
+            return
+        fi
+    fi
```

Comment:
For what do we need perl? Can we replace it with e.g. python?

Reply:
It is used as one of the attempts to properly normalize paths in the script. There is no clean way to do it in bash.

It's not a strict dependency, if not present, we use something else, but it is likely that Perl is installed and it is also needed for building OpenSSL.